user/e16: new package

digitalrane opened the pull request:
This MR adds the enlightenment X11 window manager, DR16 version 1.0.30, which is still actively maintained upstream and the latest stable version. This is maintained and still developed as and alternative to newer versions of enlightenment.

I have built and tested this locally, including launching and interacting with various applications, enabling compositing mode, and switching themes and all works well. I have tested it using `emptty` as a display manager and can confirm the session files upstream E16 provides are correctly registered when the package is installed.

This is my first contribution to cports so if there is any feedback or tips on how to contribute changes better please let me know!

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,61 @@
+pkgrel = 0
+build_style = "gnu_configure"
+configure_args = [
+    "--prefix=/usr",
+    "--sysconfdir=/etc",
+    "--mandir=/usr/share/man",
+    "--localstatedir=/var",
```
```suggestion
```
these are the defaults for gnu_configure

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,61 @@
+    "--mandir=/usr/share/man",
+    "--localstatedir=/var",
+    "--enable-sound",
+    "--enable-sound-alsa",
```
we generally don't enable alsa in things when they also support pulse/pipewire etc, unless 'alsa' means something like 'uses alsa for midi' (but not for basic audio output)

digitalrane replied:
OK that's really helpful context to have for packaging this, and for future packages. I've removed ALSA support as it's an alternate backend e16 can use to speak to ALSA devices directly. With the pulse/libpulse support sound will work just fine with ALSA disabled as long as pulseaudio or pipewire is running, so it's not an issue to disable ALSA if the preference is for pulse support to be used.

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,61 @@
+maintainer = "rane <rane+chimera@junkyard.systems>"
+license = "BSD-2-Clause"
+url = "https://enlightenment.org/e16"
+source = f"https://downloads.sourceforge.net/enlightenment/e16/{pkgver}/e16-{pkgver}.tar.gz"
```
```suggestion
source = f"$(SOURCEFORGE_SITE)/enlightenment/e16/{pkgver}/e16-{pkgver}.tar.xz"
```

digitalrane replied:
OK thanks I've changed it, that's useful to know.

nekopsykose replied:
note also the .xz on the end; they make one here and it's purely smaller :)

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,61 @@
+sha256 = "b8b7748a2c48c4c7c9758d9ad12b14f566d2bec38f2eda533e6d874f5ce9074c"
+
+
+def init_install(self):
+    self.make_install_args += ["DESTDIR=" + str(self.chroot_destdir), "install"]
```
this is the default install for gnu_configure, pretty sure it works without this

digitalrane replied:
OK thanks, that's really helpful to know. I changed it and it worked without any issues.

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,61 @@
+    "--enable-sound-alsa",
+    "--enable-sound-pulse",
+]
+hostmakedepends = ["automake", "libtool", "pkgconf"]
```
if it builds with libtool swapped with slibtool, put that there instead

digitalrane replied:
I did a rebuilt with `slibtool` and everything looks OK, so I changed it, thanks!

nekopsykose commented:
you'll also need an update.py that works to get the latest version, see other ones in the repo for examples

nekopsykose commented:
url = "https://sourceforge.net/projects/enlightenment/rss" is probably what you want for update.py

digitalrane commented:
> you'll also need an update.py that works to get the latest version, see other ones in the repo for examples

OK thanks, that's good to know. I figured that was more for things like Python library and such, but I can see the `update.py` is really super flexible and can be used for anything, so TIL. Thanks for the advice, I've added and tested an `update.py` for this package.

digitalrane commented:
thank you for the review @nekopsykose! I think i've addressed all of the feedback now, if you get a chance to review again that would be really appreciated. I tested the updated package with the discussed changes and everything seems to work well.

nekopsykose commented on user/e16/update.py:
```diff
@@ -0,0 +1,2 @@
+pattern = r"e16-(\d.\d.\d+)\.tar\.gz/"
```
```suggestion
```

the default pattern works in this case

digitalrane replied:
OK great, I've removed the patter and tested the update check, it seems to work as it did before. Changes have been pushed to this branch.

nekopsykose commented on user/e16/template.py:
```diff
@@ -0,0 +1,52 @@
+maintainer = "rane <rane+chimera@junkyard.systems>"
+license = "BSD-2-Clause"
+url = "https://enlightenment.org/e16"
+source = f"$(SOURCEFORGE_SITE)/enlightenment/e16/{pkgver}/e16-{pkgver}.tar.gz"
```
```suggestion
source = f"$(SOURCEFORGE_SITE)/enlightenment/e16/{pkgver}/e16-{pkgver}.tar.xz"
```

q66 commented on user/e16/template.py:
```diff
@@ -0,0 +1,52 @@
+    "libxxf86vm-devel",
+    "pango-devel",
+    "xbitmaps",
+    "xorgproto",
```
xorgproto is a host dependency

q66 commented on user/e16/template.py:
```diff
@@ -0,0 +1,52 @@
+]
+hostmakedepends = ["automake", "slibtool", "pkgconf"]
+makedepends = [
+    "alsa-lib-devel",
```
this should not be needed i think

q66 commented:
i ended up addressing some of the remaining nits myself since they were small things
